include/zephyr/posix/fnmatch.h

```diff
@@ -40,17 +40,33 @@
 #define FNM_NOSYS   2 /* Function not implemented. */
 #define FNM_NORES   3 /* Out of resources */
 
-#define FNM_NOESCAPE	0x01 /* Disable backslash escaping. */
-#define FNM_PATHNAME	0x02 /* Slash must be matched by slash. */
-#define FNM_PERIOD	0x04 /* Period must be matched by period. */
-#define FNM_CASEFOLD	0x08 /* Pattern is matched case-insensitive */
+#define FNM_NOESCAPE    0x01 /* Disable backslash escaping. */
+#define FNM_PATHNAME    0x02 /* Slash must be matched by slash. */
+#define FNM_PERIOD      0x04 /* Period must be matched by period. */
+#define FNM_CASEFOLD    0x08 /* Pattern is matched case-insensitive */
 #define FNM_LEADING_DIR 0x10 /* Ignore /<tail> after Imatch. */
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-int fnmatch(const char *, const char *, int);
+/**
+ * Function used to check if the given pattern is in the string, using the given flags
+ * @param pattern pattern that is matched against the string
+ * @param string string that is checked for the pattern
+ * @param flags used to signal special matching conditions
+ *        FNM_NOESCAPE	0x01 Disable backslash escaping.
+ *        FNM_PATHNAME	0x02 Slash must be matched by slash.
+ *        FNM_PERIOD	0x04 Period must be matched by period.
+ *        FNM_CASEFOLD	0x08 Pattern is matched case-insensitive
+ *        FNM_LEADING_DIR 0x10 Ignore /<tail> after Imatch.
+ * @return int
```

Comment:
Normally, we use `retval` for specific values, or `return` for a more generic description, but we typically don't use both, so I would remove the `return` bit.
```suggestion
```